src/Validator/Constraints/BonusPointsApplyValidator.php

```diff
@@ -40,11 +40,19 @@ public function validate($bonusPoints, Constraint $constraint): void
     {
         $bonusPointsStrategies = $this->bonusPointsStrategyRepository->findActiveByCalculatorType(PerOrderPriceCalculator::TYPE);
 
+        if (\count($bonusPointsStrategies) === 0) {
+            return;
+        }
+
         $eligibleBonusPointsStrategies = $this->extractOnlyEligibleStrategies($bonusPointsStrategies);
 
+        if (\count($eligibleBonusPointsStrategies) === 0) {
+            $this->context->buildViolation($constraint->messageInvalidOrderItem)->addViolation();
+        }
+
         foreach ($eligibleBonusPointsStrategies as $eligibleBonusPointsStrategy) {
             if ($bonusPoints % 100 !== 0 || $bonusPoints < 100) {
-                $this->context->buildViolation($constraint->message)->addViolation();
+                $this->context->buildViolation($constraint->messageInvalidNumber)->addViolation();
```

Comment:
maybe better name would be invalidStrategyPointsValueMessage?

Reply:
I think, better name would be invalidBonusPointsValueMessage

Reply:
👍